sources/core/Stride.Core.Design/VisualStudio/VisualStudioVersions.cs

```diff
@@ -68,102 +82,117 @@ public IDEInfo(Version version, string displayName, string installationPath, boo
         /// <summary>
         /// <c>true</c> if this IDE has a VSIX installer; otherwise, <c>false</c>.
         /// </summary>
-        public bool HasVsixInstaller => !string.IsNullOrEmpty(VsixInstallerPath) && VsixInstallerVersion != VSIXInstallerVersion.None;
+        public bool HasVsixInstaller => !string.IsNullOrEmpty(VsixInstallerPath);
+
+        /// <summary>
+        /// Thd path to the current user's Visual Studio extension installation directory.
+        /// </summary>
+        public string VsixInstallationPath { get; }
+
+        /// <summary>
+        /// Computes and verifies the location of devenv.exe relative to the given path or null if not found. 
+        /// </summary>
+        /// <param name="idePath">The path of this VS instance</param>
+        /// <returns>The full path to devenv.exe or null.</returns>
+        private string ComputeDevenvPath(string idePath)
+        {
+            var devenvPath = Path.Combine(idePath, "devenv.exe");
+            if (!File.Exists(devenvPath))
+                devenvPath = null;
+            return devenvPath;
+        }
+
+        /// <summary>
+        /// Computes and verifies the location of the VSIX installer exe relative to the given path or null if not found.
+        /// </summary>
+        /// <param name="idePath">The path of this VS instance</param>
+        /// <returns>The path of the VSIX installed for this instance or null.</returns>
+        private string ComputeVsixInstallerPath(string idePath)
+        {
+            var vsixInstallerPath = Path.Combine(idePath, "VSIXInstaller.exe");
+            if (!File.Exists(vsixInstallerPath))
+                vsixInstallerPath = null;
+            return vsixInstallerPath;
+        }
+
+        /// <summary>
+        /// Computes the installation path for Visual Studio Extensions for the current instance. 
+        /// This will always be the per-user location.
+        /// </summary>
+        /// <returns>The full path to the VSIX installation folder for this instance.</returns>
+        private string ComputeVsixInstallationPath()
```

Comment:
Adding this to support detecting whether the extension is installed or not.  Should help with the 'Reinstall' case or if the user has manually removed the extension in VS.   In the latter case, the VSIX installer will throw a nasty error message so we should just detect and skip uninstall.

Reply:
I should mention that the work to detect/skip is not in iteration 1 of this PR.   Will include it later or as a separate PR.

Reply:
Turns out the detection logic was pretty easy with my other change in place so the installer's uninstall operation will now detect if the VS2022 plugin is installed and skip if not.

Reply:
I'm pretty sure you would just need to call the VSIXInstaller with `/quiet`, so:
```cs
RunVsixInstaller(ideInfo.VsixInstallerPath, "/quiet /uninstall:Stride.VisualStudio.Package.2022");
```
Inside of `sources/tools/Stride.VisualStudio.PackageInstall/Program.cs`

Reply:
That would work but the difference is that it's still invoking VSIXInstaller which will take more than a few seconds vs. a few hundred milliseconds to check the file system and skip calling VSIXInstaller.

Reply:
I don't think it matters that much given that it's only once per installation, easier to maintain and future proof ?

Reply:
OK, it a good tradeoff: small perf degradation in some edge cases for better maintainability given Stride's limited contributor group.  I'll update and retest.

Since so much of the VS extension handling pieces are opaque to the public or explicitly not supported, Stride managing extensions on behalf of VS is probably more headache than it's worth.   As I mentioned in Discord, I believe the better direction is to publish the extension to the VS Marketplace and direct folks to install it from there.  VS has all the right support for this built in.

Reply:
Code removed and it works as expected with the VSIX installed and uninstalled.

Reply:
Regarding the VS marketplace, I do think you're right but I feel like we would need xen to chime in for this and it looks like he doesn't have a lot of time right now to do so. Something to investigate later.